guvnor-ala/guvnor-ala-wildfly-provider/src/test/java/org/guvnor/ala/wildfly/executor/tests/WildflyRuntimeTest.java

```diff
@@ -0,0 +1,171 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.ala.wildfly.executor.tests;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.maven.project.MavenProject;
+import org.arquillian.cube.CubeController;
+import org.arquillian.cube.HostIp;
+import org.arquillian.cube.requirement.ArquillianConditionalRunner;
+import org.guvnor.ala.build.maven.model.MavenBinary;
+import org.guvnor.ala.build.maven.model.impl.MavenBinaryImpl;
+import org.guvnor.ala.build.maven.model.impl.MavenProjectImpl;
+import org.guvnor.ala.build.maven.util.MavenBuildExecutor;
+import org.guvnor.ala.build.maven.util.RepositoryVisitor;
+import org.guvnor.ala.source.Source;
+import org.guvnor.ala.source.git.GitHub;
+import org.guvnor.ala.source.git.GitRepository;
+import org.guvnor.ala.wildfly.access.WildflyAppState;
+import org.guvnor.ala.wildfly.access.WildflyClient;
+import org.jboss.arquillian.junit.InSequence;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.scanner.embedder.MavenProjectLoader;
+
+import static org.junit.Assert.*;
+import org.uberfire.java.nio.file.Path;
+
+/**
+ * Test the Wildfly Provider by starting a docker image of wildfly and deploying
+ * an application there.
+ */
+@RunWith(ArquillianConditionalRunner.class)
+public class WildflyRuntimeTest {
+
+    private static final String CONTAINER = "swarm";
+    private static File tempPath;
+
+    @HostIp
+    private String ip;
+
+    @ArquillianResource
+    private CubeController cc;
+
+    @BeforeClass
+    public static void setUp() {
+        try {
+            tempPath = Files.createTempDirectory( "ooo" ).toFile();
+        } catch ( IOException e ) {
+            e.printStackTrace();
+        }
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        FileUtils.deleteQuietly( tempPath );
+    }
+
+    @Test
+    @InSequence(1)
+    public void shouldBeAbleToCreateAndStartTest() {
+        cc.create( CONTAINER );
+        cc.start( CONTAINER );
+    }
+
+    @Test
+    @InSequence(2)
+    public void waitForAppBuildTest() {
+        final GitHub gitHub = new GitHub();
+        final GitRepository repository = (GitRepository) gitHub.getRepository( "salaboy/drools-workshop", new HashMap<String, String>() {
+            {
+                put( "out-dir", tempPath.getAbsolutePath() );
+            }
+        } );
+        final Source source = repository.getSource( "master" );
+        assertNotNull( source );
```

Comment:
All the above is not the most correct way to do what you want, this is more what you're looking for:

```java
        final Optional<Source> _source = new GitConfigExecutor( new InMemorySourceRegistry() ).apply( new GitConfigImpl( tempPath.getAbsolutePath(),
                                                                                                                  "master",
                                                                                                                  "https://github.com/salaboy/drools-workshop",
                                                                                                                  "drools-workshop",
                                                                                                                  "true" ) );

        assertTrue( _source.isPresent() );
        final Source source = _source.get();

```